processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -298,38 +297,15 @@ else if ( reportErrorOnTargetObject( mapping ) ) {
 
         private void handleDefinedNestedTargetMapping(Set<String> handledTargets) {
 
-            Map<PropertyEntry, MappingOptions> optionsByNestedTarget =
-                method.getMappingOptions().groupByPoppedTargetReferences();
-            for ( Entry<PropertyEntry, MappingOptions> entryByTP : optionsByNestedTarget.entrySet() ) {
+            NestedTargetPropertyMappingHolder holder = new NestedTargetPropertyMappingHolder.Builder()
```

Comment:
and you can do something later on as well to check if sources are also nested (when a target is nested), to check for the symmetrical case.